player/player.gd

```diff
@@ -44,13 +48,13 @@ func _process(delta: float) -> void:
 	if _is_reloading:
 		update_reload(delta)
 
-func _physics_process(_delta: float) -> void:
-	if Input.is_action_just_pressed("shoot"):
+func _unhandled_input(event: InputEvent) -> void:
+	if event.is_action_pressed("shoot"):
 		if can_shoot():
 			shoot()
 		else:
 			no_ammo_audio_stream.play()
```

Comment:
this is where the bug is for getting "out of ammo" clicks when shooting. our `can_shoot` also takes the fire rate into consideration, not just the ammo

```gdscript
  if can_shoot():
    shoot()
  elif not has_ammo():
    no_ammo_audio_stream.play()

func has_ammo(): # NEW
  return _ammo > 0

func can_shoot():
  return has_ammo() and fire_timer <= 0
```

Reply:
Making a change to fix this, slightly differently tho